server/src/main/java/org/opensearch/transport/TransportService.java

```diff
@@ -544,7 +544,9 @@ public ConnectionManager.ConnectionValidator connectionValidatorForExtensionConn
     ) {
         return (newConnection, actualProfile, listener) -> {
             // We don't validate cluster names to allow for CCS connections.
-            threadPool.getThreadContext().putHeader("extension_unique_id", extensionUniqueId);
+            if (Strings.isNullOrEmpty(threadPool.getThreadContext().getHeader("extension_unique_id"))) {
```

Comment:
```suggestion
            String currentId = threadPool.getThreadContext().getHeader("extension_unique_id");
            if (currentId == null || !extensionUniqueId.equals(currentId) {
```

Reply:
Don't we have to throw an exception only when extension unique is different and add a header when the id is null?